### PR TITLE
feature: Add Login Component scaffold

### PR DESCRIPTION
**Pull request by cmejet**
*  Base Login component added, updated Login Screen to reference 
*  Added TextInput and Button Primatives as styled components
(still needs work to implement the icon with the textInput)
*  Updated component.ejs to use container instead of the old view

Expect to be able to Navigate to Login Screen From Intro and then see new Login Component with some placeholder navigation events
Screen Shot 2019-11-06 at 11 43 47 AM

### CODE REVIEW

File: template/_templates/component/new/component.ejs [hunk 1]
```diff
@@ -22,9 +22,9 @@ export const <%= formattedName %>: FC<Props> = (props = DEFAULT_PROPS) => {
   const { children } = props;
 
   return (
-    <View>
+    <Container>
       {children}
-    </View>
+    </Container>
```
Comment (rynatk) [inline template/_templates/component/new/component.ejs]: 💯

File: template/_templates/component/new/stories.ejs [hunk 2]
```diff
@@ -5,7 +5,7 @@ to: src/components/<%= h.inflection.transform(name, ['undasherize', 'classify'])
 import { storiesOf } from '@storybook/react-native';
 import React from 'react';
 
-import { <%= formattedName %> } from './component';
+import { <%= formattedName %> } from './<%= formattedName %>';
```
Comment (rynatk) [inline template/_templates/component/new/stories.ejs]: 💯

File: template/src/components/Button/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  label: string;
+  /** the callback to be invoked onPress */
+  onPress: () => void;
+}
+
+type ComponentProps = ButtonProps & BorderProps & ColorProps & SpaceProps & FlexProps & LayoutProps;
+
+const Wrapper = styled(Touchable)`
+  ${color}
+
```
Comment (rynatk) [inline template/src/components/Button/Button.tsx]:
```suggestion
```

File: template/src/components/Button/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  return (
+    <ButtonContainer
+      centerContent
+      bg={disabled ? colors.disabled : backgroundColor}
+      borderRadius={40}
+      height={50}
+      width={0.6}
+      accessibilityLabel={accessibilityLabel}
+      onPress={disabled ? null : onPressAction}
```
Comment (rynatk) [inline template/src/components/Button/Button.tsx]: sorry, 1 more tweak:

```suggestion
      disabled={disabled}
      onPress={onPressAction}
```

File: template/src/components/Login/Login.tsx [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+interface LoginProps {
+  /** the callbacks to be invoked onPress */
+  loginPress: () => void;
+  registrationPress: () => void;
+  forgotPasswordPress: () => void;
+}
+
+type ComponentProps = LoginProps & FlexProps & SpaceProps & BorderProps & ColorProps;
+
+const Divider = styled(Container)`
```
Comment (rynatk) [inline template/src/components/Login/Login.tsx]: We can remove the definition of `Divider` here

File: template/src/components/Login/Login.tsx [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+        marginTop={40}
+        label="Login"
+        color={colors.white}
+        borderRadius={5}
+        onPress={() => {
+          loginPress();
+        }}
+      />
+
+      <Divider flexDirection={'row'} marginTop={20}>
```
Comment (rynatk) [inline template/src/components/Login/Login.tsx]:
```suggestion
      <Container flexDirection={'row'} marginTop={20}>
```

File: template/src/components/TextInput/TextInput.tsx [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+type ComponentProps = TextInputProps &
+  ColorProps &
+  SpaceProps &
+  TextStyleProps &
+  TypographyProps &
+  BorderProps &
+  LayoutProps &
+  FlexProps;
+
+const Wrapper = styled(Container)`
```
Comment (rynatk) [inline template/src/components/TextInput/TextInput.tsx]: we can remove this definition fo `Wrapper`

File: template/src/components/TextInput/TextInput.tsx [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+  ${color};
+  ${layout};
+  ${space};
+  ${textStyle};
+  ${typography};
+`;
+
+// NOTE: for layout and dimensioning of TextInput, wrap it in a Container
+export const TextInput: FC<ComponentProps> = ({ topLabel, multiline, ...inputProps }) => (
+  <Wrapper grow={multiline}>
```
Comment (rynatk) [inline template/src/components/TextInput/TextInput.tsx]:
```suggestion
  <Container fill={multiline}>
```

File: template/src/components/TextInput/TextInput.tsx [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+      autoCapitalize="none"
+      underlineColorAndroid={colors.transparent}
+      selectionColor={colors.primary}
+      multiline={multiline}
+      {...inputProps}
+    />
+  </Wrapper>
+);
+
+// grow to the full width of its parent container
```
Comment (rynatk) [inline template/src/components/TextInput/TextInput.tsx]: we can remove this whole `Wrapper.defaultProps` object

File: template/src/components/TextInput/TextInput.tsx [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+
+const Wrapper = styled(Container)`
+  ${props =>
+    props.grow &&
+    `
+      flex: 1;
+    `}
+`;
+
+const TopLabel = styled(Text)``;
```
Comment (rynatk) [inline template/src/components/TextInput/TextInput.tsx]: We can remove this definition of `TopLabel`

File: template/src/components/TextInput/TextInput.tsx [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+    />
+  </Wrapper>
+);
+
+// grow to the full width of its parent container
+Wrapper.defaultProps = {
+  width: '100%',
+  grow: false,
+};
+
```
Comment (rynatk) [inline template/src/components/TextInput/TextInput.tsx]: we can remove this whole TopLabel.defaultProps object

File: template/src/components/TextInput/TextInput.tsx [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+  ${layout};
+  ${space};
+  ${textStyle};
+  ${typography};
+`;
+
+// NOTE: for layout and dimensioning of TextInput, wrap it in a Container
+export const TextInput: FC<ComponentProps> = ({ topLabel, multiline, ...inputProps }) => (
+  <Wrapper grow={multiline}>
+    {topLabel ? <TopLabel>{topLabel}</TopLabel> : null}
```
Comment (rynatk) [inline template/src/components/TextInput/TextInput.tsx]:
```suggestion
    {topLabel ? <Text color={colors.gray} fontSize={2} marginVertical={0.5}>{topLabel}</Text> : null}
```

File: template/src/components/TextInput/TextInput.tsx [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+  grow: false,
+};
+
+TopLabel.defaultProps = {
+  color: colors.gray,
+  fontSize: 2,
+  my: 0.5,
+};
+
+TextInput.defaultProps = {
```
Comment (rynatk) [inline template/src/components/TextInput/TextInput.tsx]: by leaving these here, it allows us to modify them via external props
(just for clarification of why the other 2 can be removed but this should likely stay)

File: template/src/screens/LoginScreen/LoginScreen.tsx [hunk 14]
```diff
@@ -1,16 +1,29 @@
...
 `;
 
 export function LoginScreen(props: NavigationScreenProps) {
+  const loginClick = () => {
+    props.navigation.navigate('Intro');
+  };
+
+  const registrationClick = () => {
+    props.navigation.navigate('Registration');
+  };
```
Comment (rynatk) [inline template/src/screens/LoginScreen/LoginScreen.tsx]:
```suggestion
  };
  
```

Comment (rynatk) [reply template/src/screens/LoginScreen/LoginScreen.tsx]: 😅

File: template/src/components/TextInput/TextInput.tsx (hunk 13 above)
Comment (cmejet) [reply template/src/components/TextInput/TextInput.tsx]: Thank you for this clarification! This is one of the questions I had about some of the button changes.

File: template/src/screens/LoginScreen/LoginScreen.tsx (hunk 14 above)
Comment (cmejet) [reply template/src/screens/LoginScreen/LoginScreen.tsx]: Good Catch!

**Pull request merged**